test/functional/rpc_net.py

```diff
@@ -249,11 +249,17 @@ def test_addnode_getaddednodeinfo(self):
         added_nodes = self.nodes[0].getaddednodeinfo()
         assert_equal(len(added_nodes), 1)
         assert_equal(added_nodes[0]['addednode'], ip_port)
+        # check that filtering by node works
+        self.nodes[0].addnode(node="11.22.33.44", command='add')
+        added_node = self.nodes[0].getaddednodeinfo(node=ip_port)
+        assert_equal(added_nodes, added_node)
         # check that node cannot be added again
         assert_raises_rpc_error(-23, "Node already added", self.nodes[0].addnode, node=ip_port, command='add')
         # check that node can be removed
         self.nodes[0].addnode(node=ip_port, command='remove')
-        assert_equal(self.nodes[0].getaddednodeinfo(), [])
+        added_nodes = self.nodes[0].getaddednodeinfo()
+        assert_equal(len(added_nodes), 1)
+        assert_equal(added_nodes[0]['addednode'], "11.22.33.44")
```

Comment:
instead of modifying the code here why not use `self.nodes[0].addnode(node="11.22.33.44", command='remove')` instead?

Reply:
Because I wanted to remove exactly the first added node.